// nexmo/lib/Nexmo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Credentials = require('./Credentials');

var _Credentials2 = _interopRequireDefault(_Credentials);

var _Message = require('./Message');

var _Message2 = _interopRequireDefault(_Message);

var _Voice = require('./Voice');

var _Voice2 = _interopRequireDefault(_Voice);

var _Number = require('./Number');

var _Number2 = _interopRequireDefault(_Number);

var _Verify = require('./Verify');

var _Verify2 = _interopRequireDefault(_Verify);

var _NumberInsight = require('./NumberInsight');

var _NumberInsight2 = _interopRequireDefault(_NumberInsight);

var _App = require('./App');

var _App2 = _interopRequireDefault(_App);

var _Account = require('./Account');

var _Account2 = _interopRequireDefault(_Account);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Nexmo =
/**
 * @param {Credentials} credentials - Nexmo API credentials
 * @param {string} credentials.apiKey - the Nexmo API key
 * @param {string} credentials.apiSecret - the Nexmo API secret
 * @param {object} options Additional options
 */
function Nexmo(credentials, options) {
  _classCallCheck(this, Nexmo);

  this._credentials = _Credentials2.default.parse(credentials);
  this._options = options;

  this.message = new _Message2.default(this._credentials, this._options);
  this.voice = new _Voice2.default(this._credentials, this._options);
  this.number = new _Number2.default(this._credentials, this._options);
  this.verify = new _Verify2.default(this._credentials, this._options);
  this.numberInsight = new _NumberInsight2.default(this._credentials, this._options);
  this.app = new _App2.default(this._credentials, this._options);
  this.account = new _Account2.default(this._credentials, this._options);
};

exports.default = Nexmo;
module.exports = exports['default'];
